src/Widgets/AbstractAppContainer.vala

```diff
@@ -80,6 +82,19 @@ namespace AppCenter {
             }
         }
 
+        private string? suggested_amount_ = null;
+        public string suggested_amount {
+            get {
+                if (this.package == null || this.package.component == null || updates_view) {
+                    return DEFAULT_PRICE_DOLLARS;
+                } else if (suggested_amount_ == null) {
+                    suggested_amount_ = this.package.component.get_custom_value ("x-appcenter-suggested-price");
+                }
+
+                return suggested_amount_ != null ? suggested_amount_ : DEFAULT_PRICE_DOLLARS;
+            }
+        }
+
```

Comment:
I feel like this code should be moved to `src/Widgets/HumbleButton.vala`. The default should be 0 if there is no Stripe key found. If there is one, then try to load the suggested price, and set it to 1 if non was set